Testcase meta-data fixes and enhancements

nashif opened the pull request.

nashif committed:
tests: use consistent test function names

Minor change to make test names consistent with other tests using ztest.
All test functions need to start with test_.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: ztest: style, tag and category fixes

Fix categories.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: posix: mqueue remove extra printk

Remove useless printk at the end of the tests.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: stackprot: move to ztest

Move test to use ztest instead of freestyle.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: support listing test cases

Parse all yaml file and create a list of declared testcases. This does
list the individual tests inside test projects, not only the projects
containing the tests, for example:

$ sanitycheck --list-tests  -T tests/net/socket/
 - net.socket.udp.send_recv_2_sock
 - net.socket.udp.v4_sendto_recvfrom
 - net.socket.udp.v6_sendto_recvfrom
 - net.socket.udp.v4_bind_sendto
 - net.socket.udp.v6_bind_sendto
 - net.socket.getaddrinfo_ok
 - net.socket.getaddrinfo_no_host
 - net.socket.tcp.v4_send_recv
 - net.socket.tcp.v6_send_recv
 - net.socket.tcp.v4_sendto_recvfrom
 - net.socket.tcp.v6_sendto_recvfrom
 - net.socket.tcp.v4_sendto_recvfrom_null_dest
 - net.socket.tcp.v6_sendto_recvfrom_null_dest
13 total.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

jukkar reviewed (changes requested):
Why we cannot generate the test case names directly from actual .c source file? They are mentioned in ztest_unit_test() function already so there should be no need to duplicate this information. Now the developer needs to make sure that the names are set in two places which will lead to mismatches.

nashif commented:
> Why we cannot generate the test case names directly from actual .c source file?

thought about this as well and that is a very valid point, actually most of the data was parsed from the c file using a script but decided to go this way for performance and accuracy. Will revisit.

nashif committed:
tests: kernel: style, tag, and category fixes

Fix coding style, test tags and use categories.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: net: style, tag, and category fixes

Fix coding style, test tags and use categories.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: crypto: style, tag, and category fixes

Fix coding style, test tags and use categories.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: posix: style, tag, and category fixes

Fix coding style, test tags and use categories.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ztest: support skipping tests

Some tests cant run everywhere, instead of completely dropping them, we
should report them as being skipped.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: parse test cases from source files

This parses the tests that run within a test project/application from
the source code and gives us a view of what was run, skipped and what
was blocked due to early termination of the test.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: allow unsupported tests to be skipped

Instead of completely excluding those tests, mark them as skipped and
provide an noop function that marks the test as skipped where test is
not supported.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: fix meta data of peripheral tests

Improve naming of tests.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

inakypg reviewed (commented):
LGTM
